### ISSUE TITLE
[0.9.3.5] Generators should cycle between power consumers or do brownouts in an event of power demand overload, or we need a power priority system

### ISSUE THREAD

**Issue by ThePiachu** [StrangeLoopGames/EcoIssues#22224]
I've been trying to do a big oil drilling plus manufacturing setup. When my manufacturing isn't working, I can run 20 oiljacks, but when they are, I can only run a fraction of that amount. Moreover, my waste filter turns on fora bout 10% of the time when the refineries are working.

Ideally, I would like everything to either run a bit slower in an event of a power demand overload, or cycle through the power consumers, or maybe even make a priority list of what gets to consume power first, but neither of those options seem to be the case. At the moment it looks like which machine works is determines somewhat at random.

It would be nice to have these options:

1) Cycling power consumers
Make a list of power consumers in the network. Turn them on one by one until max power is reached. Remember where the iteration ended. Next power tick, start the iteration at the first item that didn't get power and go from there. This way every machine would get some power and things like waste filter would clear through easily.

2) Brownouts
Every machine is penalised in speed by the percent of available power (see Factorio). This way everything would run in parallel, but slower. Good for when you have some small burst power requirements that you don't need to baby.

3) Power priority
Have a list of power consumers in an electric network (see #22199 ). The list should be its own screen you can drag and drop to order things around. It could have a few levels of priority so you can easily move the same machines around ("waste filter priority 0, workbenches priority 1, 20 pumpjacks priority 9".

Relevant issues - #15869 , #19720
